extensions-utils/src/download_dependencies.rs

```diff
@@ -0,0 +1,44 @@
+use flate2::read::GzDecoder;
+use std::fs::File;
+use std::path::Path;
+use std::{env, fs, io::copy};
+
+pub fn download_ic_binary(replica_rev: &str, binary_name: &str, renamed_binary_name: &str) {
+    let arch = match std::env::consts::ARCH {
+        "x86_64" => "x86_64",
+        "aarch64" => "x86_64", // let's rely on rosetta2 for now, since sns-cli is not available for arm64
+        _ => panic!("Unsupported architecture"),
+    };
+    let os = match std::env::consts::OS {
+        "macos" => "darwin",
+        "linux" => "linux",
+        // "windows" => "windows", // unsupported till dfx supports windows
+        _ => panic!("Unsupported OS"),
+    };
+
+    let url = format!(
+        "https://download.dfinity.systems/ic/{replica_rev}/openssl-static-binaries/{arch}-{os}/{binary_name}.gz",
+        arch = arch,
+        os = os,
+        binary_name = binary_name,
+    );
+    println!("Downloading {}", url);
+
+    let resp = reqwest::blocking::get(&url).expect("Request failed");
```

Comment:
Please add retry logic

Reply:
Recommendation: use the backoff crate, for example https://github.com/dfinity/sdk/pull/3255/files